src/backend/api/mirrors_update.py

```diff
@@ -249,17 +249,19 @@ async def mirror_available(
                 async with http_session.get(
                     check_url,
                     headers=HEADERS,
-                    timeout=45,
+                    timeout=30,
                 ) as resp:
                     await resp.text()
-                    if resp.status == 200:
-                        # if mirror has at least one valid version/arch combo
-                        logger.info(
-                            'Mirror "%s" is available',
-                            mirror_name,
+                    if resp.status != 200:
+                        # if mirror has no valid version/arch combos it is dead
+                        logger.error(
+                            'Mirror "%s" has one or more invalid repositories',
+                            mirror_name
                         )
-                        return mirror_name, True
-            except (ClientError, asyncio.TimeoutError) as err:
+                        return mirror_name, False
+            except (ClientError, asyncio.exceptions.TimeoutError) as err:
+                if type(err) == asyncio.exceptions.TimeoutError:
+                    err = type(err)
```

Comment:
For what is this conversion? Why do we not use variable `err` as is?

Reply:
Because `TimeoutError` has a blank error body so it simply results in something like this in the logs:

> Mirror "mirror.xyz.com" is not available for version "8.5" and repo path "PowerTools/debug/aarch64/" because ""

Instead of the blank reason now we get a reasonable entry such as:

> <class 'asyncio.exceptions.TimeoutError'>

in the "because" field so it is at least clear what happened.